enterprise/audit/table.go

```diff
@@ -4,8 +4,26 @@ import (
 	"reflect"
 
 	"github.com/coder/coder/coderd/database"
+	"github.com/coder/coder/codersdk"
 )
 
+// This mapping creates a relationship between an Auditable Resource
+// and the Audit Actions we track for that resource.
+// It is important to maintain this mapping when adding a new Auditable Resource to the
+// AuditableResources map (below) as our documentation - generated in scripts/auditdocgen/main.go -
+// depends upon it.
+var AuditActionMap = map[string][]string{
+	"GitSSHKey":          {string(codersdk.AuditActionCreate)},
+	"OrganizationMember": {},
+	"Organization":       {},
+	"Template":           {string(codersdk.AuditActionWrite), string(codersdk.AuditActionDelete)},
+	"TemplateVersion":    {string(codersdk.AuditActionCreate), string(codersdk.AuditActionWrite)},
+	"User":               {string(codersdk.AuditActionCreate), string(codersdk.AuditActionWrite), string(codersdk.AuditActionDelete)},
+	"Workspace":          {string(codersdk.AuditActionCreate), string(codersdk.AuditActionWrite), string(codersdk.AuditActionDelete)},
+	"WorkspaceBuild":     {string(codersdk.AuditActionStart), string(codersdk.AuditActionStop)},
+	"AuditableGroup":     {string(codersdk.AuditActionCreate), string(codersdk.AuditActionWrite), string(codersdk.AuditActionDelete)},
+}
+
```

Comment:
@coadler if you can think of a better way to get this relationship rather than maintaining this map, I'm all ears.

Reply:
I think this is the best way currently!

Reply:
Not blocking this PR, but would be beneficial as a next step:

What would make sense now is refactoring the rest of the codebase to use only this map to figure out the `AuditAction`. Otherwise, it will go out of sync relatively quickly. Alternatively, we can implement a unit test that checks if the map is in-sync.

Reply:
@mtojek @coadler  When we make an audit request for a resource, [we pass an action](https://github.com/coder/coder/blob/main/coderd/templates.go#L91). I could grab the action from this new map. This would be a quick update, but would touch every audit request in the code base. 
Would this be too heavy-handed an association?

Otherwise, I can explore the unit test that looks at the keys in `AuditableResources` and ensures there are identical keys in `AuditActionMap`.

Let me know which approach seems better.